smart_open/__init__.py

```diff
@@ -27,11 +27,44 @@
 #
 # Prevent regression of #474 and #475
 #
-logging.getLogger(__name__).addHandler(logging.NullHandler())
+logger = logging.getLogger(__name__)
+logger.addHandler(logging.NullHandler())
 
 from smart_open import version  # noqa: E402
 from .smart_open_lib import open, parse_uri, smart_open, register_compressor  # noqa: E402
-from .s3 import iter_bucket as s3_iter_bucket  # noqa: E402
+
+deprecated_warning = """smart_open.s3_iter_bucket is deprecated and will stop functioning
```

Comment:
This looks like an internal constant, so it's better to explicitly name it like one.

```suggestion
_WARNING = """smart_open.s3_iter_bucket is deprecated and will stop functioning
```